pkg/nsx/services/securitypolicy/firewall.go

```diff
@@ -1214,3 +1203,59 @@ func (service *SecurityPolicyService) getVPCInfo(spNameSpace string) (*common.VP
 	}
 	return &vpcInfo[0], nil
 }
+
+func (service *SecurityPolicyService) getPolicyAppliedGroupByCRUID(indexScope, uid string) *model.Group {
+	groups := service.groupStore.GetByIndex(indexScope, uid)
+	if len(groups) == 0 {
+		return nil
+	}
+	for _, group := range groups {
+		ruleTags := filterRuleTag(group.Tags)
+		if len(ruleTags) == 0 {
+			return group
+		}
+	}
+	return nil
+}
+
+func (service *SecurityPolicyService) getAppliedGroupByRuleId(createdFor, uid string, ruleIdTag string) *model.Group {
+	indexScope := common.TagValueScopeSecurityPolicyUID
+	if createdFor == common.ResourceTypeNetworkPolicy {
+		indexScope = common.TagScopeNetworkPolicyUID
+	}
+
+	if ruleIdTag == "" {
+		return service.getPolicyAppliedGroupByCRUID(indexScope, uid)
+	}
+
+	groups := service.groupStore.GetByIndex(common.TagScopeRuleID, ruleIdTag)
+	if len(groups) == 0 {
+		return nil
+	}
+
+	for _, group := range groups {
+		if strings.Contains(*group.Id, common.TargetGroupSuffix) {
```

Comment:
@timdengyun Do you think I shall use `fmt.String("_%s", common.TargetGroupSuffix)` (old format) and/or `fmt.String("-%s_", common.TargetGroupSuffix)` in string compare? So does function `getPeerGroupByRuleId` . Otherwise it cause detect failure if the user created CR name (after truncation) has common.TargetGroupSuffix

Reply:
For handling this corner case,  I feel that we should `fmt.String("_%s", common.TargetGroupSuffix)`
or` fmt.String("-%s_", common.TargetGroupSuffix)` since CR name can not have _.
since we have two formats(old and new), I feel that we can only use the new one and the old group will be updated as the new format.

Reply:
I feel that it's ok, anyway, we will recreate rule id as well if we do the similar change, right?